Oauth device code grant is not working with OIDC provider

**Description**

The `cosign sign`  or `cosign sign-blob` is not working for device code flow.  `invalid_client` error is received from the OIDC provider (Keycloak in current case).

```
$ cosign -d sign-blob ../test/file.txt  --oidc-client-id='trusted-artifact-signer' --fulcio-auth-flow=device
Using payload from: ../test/file.txt
Generating ephemeral keys...
Retrieving signed certificate...
Enter the verification code JRSW-HFYD in your browser at: https://keycloak-rhbk-operator.xxxxxx/realms/tas-realm/device?user_code=JRSW-HFYD
Code will be valid for 600 seconds
Error: signing ../test/file.txt: getting key from Fulcio: retrieving cert: unexpected error in device flow: invalid_client
main.go:74: error during command execution: signing ../test/file.txt: getting key from Fulcio: retrieving cert: unexpected error in device flow: invalid_client
```
It seems the device code flow in sigstore is not as per the OAuth 2.0 Device Authorization Grant [specification](https://datatracker.ietf.org/doc/html/rfc8628#section-3.4), where the access token request is missing `client_id` parameter. client_id is a required parameter in Device Access Token Request.


The issue can be reproduced by using `--fulcio-auth-flow=device` and with a OIDC provider ( that supports device code grant , example Keycloak) configured.

**Version**
```
GitVersion:    v2.2.4
GitCommit:     fb651b4ddd8176bd81756fca2d988dd8611f514d
GitTreeState:  clean
BuildDate:     2024-04-10T21:57:27Z
GoVersion:     go1.21.8
Compiler:      gc
Platform:      linux/amd64
```

--- a/pkg/oauthflow/device.go
+++ b/pkg/oauthflow/device.go
@@ -135,8 +135,9 @@ func (d *DeviceFlowTokenGetter) deviceFlow(p *oidc.Provider, clientID, redirectU
 		// Some providers use a secret here, we don't need for sigstore oauth one so leave it off.
 		data := url.Values{
 			"grant_type":    []string{"urn:ietf:params:oauth:grant-type:device_code"},
+			"client_id":     []string{clientID},
 			"device_code":   []string{parsed.DeviceCode},
-			"scope":         []string{"openid", "email"},
+			"scope":         []string{"openid email"},
 			"code_verifier": []string{pkce.Value},
 		}
 

